sdk/tables/Azure.Data.Tables/src/DynamicTableEntity.cs

```diff
@@ -0,0 +1,221 @@
+        public byte[] GetBinary(string key) => GetValue<byte[]>(key);
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="String"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public bool GetBoolean(string key) => GetValue<bool>(key);
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="DateTime"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public DateTime GetDateTime(string key) => GetValue<DateTime>(key);
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="Double"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public double GetDouble(string key) => GetValue<double>(key);
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="Guid"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public Guid GetGuid(string key) => GetValue<Guid>(key);
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="Int32"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public int GetInt32(string key) => GetValue<int>(key);
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="Int64"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public long GetInt64(string key) => GetValue<long>(key);
+
+        /// <summary>
+        /// Set a document property.
+        /// </summary>
+        /// <param name="key">The property name.</param>
+        /// <param name="value">The property value.</param>
+        private protected void SetValue(string key, object value)
+        {
+            Argument.AssertNotNullOrEmpty(key, nameof(key));
+
+            if (_properties.TryGetValue(key, out object existingValue))
+            {
+                EnforceType(existingValue.GetType(), value.GetType());
```

Comment:
Nit - what happens if either value is null?

Reply:
`existingValue` can't be null because `_properties.TryGetValue` will return false if null or no value is found. And for `value`, that did come up with a error. While fixing this, I had a question about how we should deal with a scenario where `Guid value = null`, but the `existingValue` is of type string, for example. Should this matter? If not, then I can do a simple null check. 

Currently, `SearchDocument` doesn't care about the types when setting the value. And in Track 1 Tables, they got around checking the type of a null value by having an `EntityProperty` class that stores the enum `EdmType`.

Reply:
`TryGetValue` will return `true` for `null` values:

```C#
Dictionary<string, string> dict = new Dictionary<string, string>();
dict["Foo"] = "hello";
dict["Bar"] = null;

string val = null;
Console.WriteLine("Foo: " + (dict.TryGetValue("Foo", out val) ? val : "undefiend"));
Console.WriteLine("Bar: " + (dict.TryGetValue("Bar", out val) ? val : "undefiend"));
Console.WriteLine("Baz: " + (dict.TryGetValue("Baz", out val) ? val : "undefiend"));

// Prints:
// 
// Foo: hello
// Bar:
// Baz: undefiend
```

I think it might make sense to support type coercion between the primitives we know about.  I think it's fine to check in like this though and follow up with the feature crew about how we're handling this across languages.

Reply:
tracking the `null` errors here #13658